### PR TITLE
build: enable local development setup with Docker and PostgreSQL

### PR DESCRIPTION
**Pull request by MartiinWalsh**
## Description

This PR introduces a new feature to enable running the API locally with a PostgreSQL database. The following changes have been made:

**Addition of Dockerfile:** A Dockerfile has been added to the root of the repository, which outlines the steps to create a Docker container for the PostgreSQL database. It configures the database with SSL for secure connections and sets up necessary parameters such as user, password, and database name. The Dockerfile also includes steps to add an SSL certificate and key to the container.

**Addition of init.sh script**: An initialisation script is introduced that configures the PostgreSQL instance to enable SSL on startup.

**Documentation Update:** The README file has been updated with instructions for building and running the Docker image locally, helping developers set up a local PostgreSQL database for development purposes.

These changes aim to provide a consistent development environment and simplify the setup process for developers.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes: #204 
## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [x] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
Thanks for the contribution @MartiinWalsh. I will give this a spin locally later today.

**Review by bdougie — APPROVED** [review]
This is beautiful. 

I found your README and Dockerfile pretty and easy to follow. note we do have plans to simplify the local dev experience, but this setup unblocks quite a few folks trying to contribute and hitting the local .env hurdle.

**Comment by MartiinWalsh** [comment]
Thank you for the positive feedback @bdougie, I really appreciate it. I'm happy to hear this will help unblock others.

I think it's great that there are plans to further simplify the local dev experience. Please feel free to reach out if there's any way I can contribute to those efforts in the future.

Thanks again!

File: init.sh
Comment (jpmcb) [inline init.sh:1]: Can we move this file somewhere else? An `init.sh` at the root of the project makes me think it's the main entry point for the code/api.

I've used `hack/` in the past to make it pretty clear that it is _not_ production usable but curious what others think.

File: Dockerfile.local
Comment (jpmcb) [inline Dockerfile.local:1]: And maybe the same goes for this file: might make sense in a `hack/` or `dev/` directory? And I think a filename like `Dockerfile.local-postgres` makes more sense to me but that's more of a nit.

**Review by jpmcb — APPROVED** [review]
Wow this is fantastic. I ran into something similar on the pizza service setting up a local postgres (but ended up opting for a more complex local kubernetes deployment). Thanks so much for this contribution!!

**Comment by MartiinWalsh** [comment]
Hey @jpmcb,

Thanks so much for your feedback!

Based on your suggestions, I've moved `init.sh` and `Dockerfile.local` to a `dev` directory. `init.sh` is now `init-docker-ssl.sh` and `Dockerfile.local` is now `Dockerfile.local-postgres` to better reflect their purpose.

**Comment by brandonroberts** [comment]
Thanks @MartiinWalsh, this is great!

**Pull request merged**